core/src/main/java/org/apache/spark/shuffle/external/ExternalShuffleDataIO.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.spark.shuffle.external;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.SparkEnv;
+import org.apache.spark.network.netty.SparkTransportConf;
+import org.apache.spark.network.util.TransportConf;
+import org.apache.spark.shuffle.api.ShuffleDataIO;
+import org.apache.spark.shuffle.api.ShuffleReadSupport;
+import org.apache.spark.shuffle.api.ShuffleWriteSupport;
+import org.apache.spark.SecurityManager;
+import org.apache.spark.util.Utils;
+
+public class ExternalShuffleDataIO implements ShuffleDataIO {
+
+    private static final String SHUFFLE_SERVICE_PORT_CONFIG = "spark.shuffle.service.port";
+    private static final String DEFAULT_SHUFFLE_PORT = "7337";
+
+    private final SparkConf sparkConf;
+    private final TransportConf conf;
+    private final SecurityManager securityManager;
+    private final String hostname;
+    private final int port;
+    private final String execId;
+
+    public ExternalShuffleDataIO(
+            SparkConf sparkConf,
+            int execId) {
```

Comment:
I think `execId` is not used?

Reply:
yea i haven't implemented the ESS server side yet. I'll see if I can remove it altogether when I figure out that code